### PR TITLE
feat: snappy decompression

### LINKED ISSUE #9: snappy decompression
would you accept PR implementing https://google.github.io/snappy/

or should I implement it in a separate processor?

### PR DESCRIPTION
**Pull request by ondrejhlavacek**
- support for `snappy` compression via https://github.com/andrix/python-snappy (fixes #9)
- use `escapeshellarg` for cli commands arguments

### CODE REVIEW

File: src/Decompressor/DecompressorSnappy.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+use Symfony\Component\Process\Process;
+
+class DecompressorSnappy extends BaseDecompressor implements DecompressorInterface
+{
+    public function decompress(SplFileInfo $sourceFile): void
+    {
+        try {
+            $baseName = $sourceFile->getBasename('.snappy');
+            $destinationPath = $this->getDestinationPath($sourceFile);
+            var_dump('python3 -m snappy -d ' . escapeshellarg($sourceFile->getPathname()) . ' ' . escapeshellarg($destinationPath . '/'  .$baseName));
```
Comment (odinuv) [inline src/Decompressor/DecompressorSnappy.php]: catch :)

**Review by odinuv — APPROVED** [review]
neslo by udelat, aby to nevypisovalo trace? raceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.5/dist-packages/snappy/__main__.py", line 89, in <module>
    cmdline_main()
  File "/usr/local/lib/python3.5/dist-packages/snappy/__main__.py", line 85, in cmdline_main
    method(args.infile, args.outfile, **additional_args)
  File "/usr/local/lib/python3.5/dist-packages/snappy/hadoop_snappy.py", line 207, in stream_decompress
    start_chunk=start_chunk
  File "/usr/local/lib/python3.5/dist-packages/snappy/snappy.py", line 323, in stream_decompress
    decompressor.flush()  # makes sure the stream ended well
  File "/usr/local/lib/python3.5/dist-packages/snappy/hadoop_snappy.py", line 181, in flush
    raise UncompressError("chunk truncated")

File: src/Decompressor/DecompressorSnappy.php (hunk 1 above)
Comment (ondrejhlavacek) [reply src/Decompressor/DecompressorSnappy.php]: 🎉

**Comment by ondrejhlavacek** [comment]
@odinuv nenapadá tě jak to udělat? našel jsem akorát toto - https://stackoverflow.com/questions/27674602/hide-traceback-unless-a-debug-flag-is-set

**Comment by odinuv** [comment]
hmm, no tak to vypada, ze je spis na bugreport na clicko :(

**Comment by ondrejhlavacek** [comment]
https://github.com/andrix/python-snappy/issues/67

**Pull request merged**